### PR TITLE
feat: revoke access with optional resource name and service name

### PR DESCRIPTION
**Pull request by bmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-334

### CODE REVIEW

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/internal/repositories/AccessManagementRepository.java [hunk 1]
```diff
@@ -34,9 +34,7 @@
         + "access_management.resource_id = :resourceId "
         + "and access_management.accessor_id = :accessorId "
         + "and access_management.accessor_type = cast(:accessorType as accessor_type) "
-        + "and access_management.service_name = :resourceDefinition.serviceName "
-        + "and access_management.resource_type = :resourceDefinition.resourceType "
```
Comment (abhijitdiwan) [inline am-lib/src/main/java/uk/gov/hmcts/reform/amlib/internal/repositories/AccessManagementRepository.java]: Can't we do some thing like :relationship is null or access_management.relationship = :relationship for option service_name and resource_type?

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java [hunk 2]
```diff
@@ -123,9 +123,9 @@ public void grantExplicitResourceAccess(@NotNull @Valid ExplicitAccessGrant acce
...
      */
     @AuditLog("explicit access revoked by '{{mdc:caller}}' to resource '{{accessMetadata.resourceId}}' defined as "
-        + "'{{accessMetadata.resourceDefinition.serviceName}}|{{accessMetadata.resourceDefinition.resourceType}}|"
-        + "{{accessMetadata.resourceDefinition.resourceName}}' from accessor '{{accessMetadata.accessorId}}' "
-        + "with relationship '{{accessMetadata.relationship}}': {{accessMetadata.attribute}}")
+        + "'{{accessMetadata.serviceName}}|{{accessMetadata.resourceType}}|{{accessMetadata.resourceName}}' "
+        + "from accessor '{{accessMetadata.accessorId}}' with relationship '{{accessMetadata.relationship}}': "
+        + "{{accessMetadata.attribute}}")
     public void revokeResourceAccess(@NotNull @Valid ExplicitAccessMetadata accessMetadata) {
         jdbi.useExtension(AccessManagementRepository.class,
```
Comment (abhijitdiwan) [inline am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: not able to see any test case checks optional resource name and service name either in Lib or API.

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/InvalidArgumentsProvider.java [hunk 3]
```diff
@@ -37,8 +41,9 @@
         List<Arguments> combinations = new ArrayList<>();
 
         Type[] parameterTypes = context.getRequiredTestMethod().getGenericParameterTypes();
+        Annotation[][] annotations = context.getRequiredTestMethod().getParameterAnnotations();
```
Comment (abhijitdiwan) [inline am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/InvalidArgumentsProvider.java]: @ValidAttributeSecurityClassification **@exampleAnotherAnnotation**
        Map<JsonPointer, SecurityClassification> attributeSecurityClassifications if we have multiple Annotations for field will  this code works?

Comment (bmcts) [reply am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/InvalidArgumentsProvider.java]: 👍 , can handle multiple annotations as discussed

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java (hunk 2 above)
Comment (bmcts) [reply am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: 👍 , added integration test for null resource name and service name

**Review by abhijitdiwan — APPROVED** [review]

**Pull request merged**